### PR TITLE
feat 2.6.0 complex consts

### PR DESCRIPTION
**Pull request by TAdev0**
#596

- - -
This change is [Reviewable](https://reviewable.io/reviews/cairo-book/cairo-book/692)

### CODE REVIEW

**Comment by TAdev0** [comment]
@enitrat  not sure that mentioning complex consts (struct enum, fixed size array) in data types chapter is ideal, what do you think? It might be good though, as it is just a small quotation that is it possible to do that

File: src/ch02-01-variables-and-mutability.md [hunk 1]
```diff
@@ -108,23 +108,34 @@ Constants can only be declared in the global scope, which makes
...
 Nonetheless, it is possible to use the `consteval_int!` macro to create a `const` variable that is the result of some computation:
 
 ```rust, noplayground
-const ONE_HOUR_IN_SECONDS: u32 = consteval_int!(60 * 60);
+{{#include ../listings/ch02-common-programming-concepts/no_listing_00_consts/src/lib.cairo:consteval_const}}
 ```
 
 We will dive into more detail about macros in the [dedicated section](./ch11-06-macros.md).
 
+We can also declare more complexe `const` variables, including structs and enums that we will discuss in later chapters:
```
Comment (enitrat) [inline src/ch02-01-variables-and-mutability.md:127]: Let's think about it the other way: which types cannot be instanciated as constants? 

My expectation is that I can create a const value of any type. Previously this was not the case and might have come surprising to the user. Now that we can have struct / enum variants as constants I think we should not enumerate what type you can create constants out of: the idea is just to introduce the _concept_ of const variables, which should transitively be applicable to all relevant types

Comment (TAdev0) [reply src/ch02-01-variables-and-mutability.md]: what do you think now?

**Review by enitrat — APPROVED** [review]
great!

**Pull request merged**